### PR TITLE
Avk003/masterchef params

### PR DESCRIPTION
**Pull request by avkr003**
Key Changes:
 - Removing redundant members from masterchef params, added it's test case
 - adding validation for ProtocolRevenueAddress BlockedAddress check in the keeper and before initializing genesis (based on Halborn  report)
 - Local testing done for migration

### CODE REVIEW

File: proto/elys/masterchef/incentive.proto [hunk 1]
```diff
@@ -6,17 +6,25 @@ import "google/protobuf/timestamp.proto";
...
+
 message IncentiveInfo {
-    // reward amount in eden for 1 year
-	string eden_amount_per_year = 1 
-        [ (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.nullable) = false];
-    // starting block height of the distribution
-    string distribution_start_block = 2
-        [ (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.nullable) = false];
-    // distribution duration - block number per year
-	string total_blocks_per_year = 3
```
[the remark this answers is not in the document]
Comment (avkr003) [reply proto/elys/masterchef/incentive.proto:17]: Yes, I checked this. We use total blocks per year from the parameters module.

File: x/masterchef/keeper/abci.go [hunk 2]
```diff
@@ -123,21 +122,17 @@ func (k Keeper) ProcessUpdateIncentiveParams(ctx sdk.Context) {
 			continue
 		}
 
-		totalBlocksPerYear := sdk.NewInt(int64(inflation.EndBlockHeight - inflation.StartBlockHeight + 1))
+		totalBlocksPerYear := inflation.EndBlockHeight - inflation.StartBlockHeight + 1
```
[the remark this answers is not in the document]
Comment (avkr003) [reply x/masterchef/keeper/abci.go:125]: Didn't get you. I removed the type conversion from int64 to sdk.Int as int64 was good enough

**Review by amityadav0 — APPROVED** [review]

**Comment by amityadav0** [comment]
Please check unit tests, they are failing

**Comment by avkr003** [comment]
Done. It was majorly dependent on one of the other PRs

**Pull request merged**